app/src/main/java/org/mozilla/focus/greenDAO/DownloadInfo.java

```diff
@@ -32,26 +33,32 @@ public DownloadInfo(long downloadId,String fileName){
             double size = cursor.getDouble(cursor.getColumnIndex(DownloadManager.COLUMN_TOTAL_SIZE_BYTES));
             long timeStamp = cursor.getLong(cursor.getColumnIndex(DownloadManager.COLUMN_LAST_MODIFIED_TIMESTAMP));
 
-            Uri = cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_LOCAL_URI));
-            String extension = MimeTypeMap.getFileExtensionFromUrl(Uri);
+            MediaUri = cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_MEDIAPROVIDER_URI));
+            FileUri = cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_LOCAL_URI));
+            String extension = MimeTypeMap.getFileExtensionFromUrl(FileUri);
             MimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension);
-            cursor.close();
 
             Status = statusConvertStr(status);
             Size = convertByteToReadable(size);
             Date = convertMillis(timeStamp);
         }
 
+        cursor.close();
```

Comment:
Since we're targeting API 19+ try-with-resource is also available. https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html